sdk/src/test/kotlin/com/pulumi/kotlin/TestUtils.kt

```diff
@@ -1,9 +1,56 @@
 package com.pulumi.kotlin
 
 import com.pulumi.core.Output
+import com.pulumi.kotlin.options.ResourceTransformation
+import com.pulumi.resources.Resource
+import com.pulumi.resources.ResourceTransformation.Args
+import com.pulumi.resources.ResourceTransformation.Result
+import io.mockk.every
+import io.mockk.mockkClass
+import io.mockk.mockkObject
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Assertions.fail
+import org.junit.jupiter.api.assertAll
+import kotlin.reflect.KClass
 
 internal fun <T> extractOutputValue(output: Output<T>?): T? {
     var value: T? = null
     output?.applyValue { value = it }
     return value
 }
+
+internal fun <T : KotlinResource, R : Resource> mockKotlinResource(
+    kotlinResourceToMock: KClass<T>,
+    javaResourceEquivalentToMock: KClass<R>,
+): T {
+    val mockedKotlinResource = mockkClass(kotlinResourceToMock)
+    val mockedJavaResource = mockkClass(javaResourceEquivalentToMock)
+
+    every {
+        mockedKotlinResource.underlyingJavaResource
+    } returns mockedJavaResource
+
+    mockkObject(GlobalResourceMapper)
+    every {
+        GlobalResourceMapper.tryMap(any<Resource>())
```

Comment:
We mock global object, so can we somehow improve this to reduce surprises when someone uses it without unmocking between every test?

I mean, we should probably only do this: `GlobalResourceMapper.tryMap(mockedJavaResource)`? (It didn't actually work for me and I didn't dig much.)